aws/cloudformation-templates/_template.yaml

```diff
@@ -0,0 +1,302 @@
+
+  CreateElasticsearchServiceLinkedRole:
+    Type: String
+    Description: >
+      If your account already has an IAM Role named 'AWSServiceRoleForAmazonElasticsearchService', select 'No'. Otherwise, select 'Yes' and one will be created automatically.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'Yes'
+
+  SourceDeploymentType:
+    Type: String
+    Description: >
+      The Retail Demo Store deployment includes a CodePipeline configuration that will build and deploy Retail Demo Store 
+      microservices to ECS when a change is detected in the source repository. The source repository for CodePipeline 
+      can be configured to use your personal GitHub repository or CodeCommit. Use GitHub if you're actively developing 
+      against Retail Demo Store in your own fork. CodeCommit is useful when you just want to get up and going quickly for 
+      a demo or evaluation or for workshop scenarios, such as Event Engine, where you want attendees to have their own 
+      source repositories provisioned.
+    AllowedValues:
+      - 'GitHub'
+      - 'CodeCommit'
+    Default: 'CodeCommit'
+
+  GitHubRepo:
+    Type: String
+    Description: Name of Retail Demo Store GitHub repository in your GitHub account.
+    Default: retail-demo-store
+
+  GitHubBranch:
+    Type: String
+    Description: Name of GitHub branch to link to this Retail Demo Store deployment.
+    Default: master
+
+  GitHubToken:
+    Type: String
+    Description: GitHub Personal Access Token for your GitHub account. Be sure that your token has the "repo", "repo:status", and "admin:repo_hook" permission scopes.
+    NoEcho: true
+
+  GitHubUser:
+    Type: String
+    Description: Your GitHub username.
+
+  PreIndexElasticsearch:
+    Type: String
+    Description: >
+      Deploy Lambda function that automatically indexes the Retail Demo Store products in Elasticsearch. Otherwise, select 'No' if you would 
+      prefer to complete this process yourself by stepping through the Search workshop included in your deployment as a Jupyter notebook in SageMaker.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+
+  PreCreatePersonalizeCampaign:
+    Type: String
+    Description: >
+      Deploy Lambda function that automatically builds solutions and launches Personalize campaigns. Otherwise, select 'No' if you would 
+      prefer to complete this process yourself by stepping through the Personalization workshop included in your deployment as a Jupyter notebook 
+      in SageMaker.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+
+  AmplitudeApiKey:
+    Type: String
+    Description: Amplitude API key for integrated product analytics and A/B testing results (optional).
+    NoEcho: true
+
+  OptimizelySdkKey:
+    Type: String
+    Description: Optimizely SDK key for experimentation (optional).
+    NoEcho: true
+
+  ParentStackName:
```

Comment:
Is there a need to pass this as an additional parameter?
Could we maybe use psuedo parameters like AWS::?
More info [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html)

Reply:
Region could be received in any nested stack via pseudo parameter, but Parent stack name will be change every time when we use nested stack (we have many nested in nested ).

Reply:
Makes sense.